File: 392-is-subsequence/392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if (s.size() == 0)
            return true;
        if (s.size() > t.size())
            return false;
        int start = 0, end = 0;
        while (end < t.size()) {
            if (t[end] == s[start]) {
                start++;
                if (start == s.size())
                    return true;
            }
            end++;
        }
        return false;
    }
};

File: 392-is-subsequence/392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        if (s.length() == 0)
            return true;
        if (s.length() > t.length())
            return false;
        int start = 0, end = 0;
        
        while (end < t.length()) {
            if (t.charAt(end) == s.charAt(start)) {
                start++;
                if (start == s.length())
                    return true;
            }
            end++;
        }
        return false;
    }
}